File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0;
        int right = numbers.size() - 1;
        int sum;

        while (left < right)
        {
            sum = numbers[left] + numbers[right];

            if (sum < target)
            {
                left++;
            }
            else if (sum > target)
            {
                right--;
            }
            else
            {
                break;
            }
        }

        return {left + 1, right + 1};
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = [1] * len(nums)

        # Get cumulative product of all numbers to left of current index
        # starting at index 1
        curr_product = 1
        for i in range(1, len(nums)):
            curr_product = curr_product * nums[i - 1]
            result[i] = curr_product

        # Get cumulative product of all numbers to right of current index
        # starting at index before last
        curr_product = 1
        for i in range(len(nums) - 1, 0, -1):
            curr_product = curr_product * nums[i]
            result[i - 1] = result[i - 1] * curr_product

        return result
